test/functional/main_test.go

```diff
@@ -21,31 +23,83 @@ const (
 	testOutputFileName   = "test-output.txt"
 )
 
-func captureOutput(f func()) string {
-	r, w, err := os.Pipe()
+func teeOutput(f func()) string {
+
+	var output bytes.Buffer
+
+	stdoutReader, stdoutWriter, err := os.Pipe()
 	if err != nil {
 		panic(err)
 	}
 
-	stdout := os.Stdout
-	os.Stdout = w
+	stderrReader, stderrWriter, err := os.Pipe()
+	if err != nil {
+		panic(err)
+	}
+
+	originalStdout := os.Stdout
+	os.Stdout = stdoutWriter
 	defer func() {
-		os.Stdout = stdout
+		os.Stdout = originalStdout
 	}()
 
-	stderr := os.Stderr
-	os.Stderr = w
+	originalStderr := os.Stderr
+	os.Stderr = stderrWriter
 	defer func() {
-		os.Stderr = stderr
+		os.Stderr = originalStderr
 	}()
 
+	var wg sync.WaitGroup
+
+	// this function will keep reading
+	// from the piped stdout/stderr and write
+	// to the original stdout/stderr
+	t := func(r, w *os.File) {
+		buf := make([]byte, 4096)
+		for true {
+			l, err := r.Read(buf)
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				panic(err)
+			}
+			if l == 0 {
+				break
+			}
+
+			_, err = w.Write(buf[:l])
+			if err != nil {
+				panic(err)
+			}
+
+			_, err = output.Write(buf[:l])
+			if err != nil {
+				break
+			}
+		}
+
+		wg.Done()
+	}
+
+	wg.Add(2)
+
+	go t(stdoutReader, originalStderr)
```

Comment:
should the second parameter here be `originalStdout`?

Reply:
Yes right, thanks for catching it